### PR TITLE
Fix MSVC test failures.

### PR DESCRIPTION
**Pull request by wmaxey**
This commit chain will include fixes to address failures detected across several versions of MSVC. I've used the compilers listed below that I believe are major releases for MSVC. 

## MSVC versions that are clean:
- [x] 14.12
- [x] 14.13
- [x] 14.16
- [x] 14.22
- [x] 14.24 
- [x] 14.27 
- [x] 14.29
- [x] 14.30

### CODE REVIEW

File: .upstream-tests/test/std/utilities/time/time.cal/time.cal.day/time.cal.day.members/plus_minus_equal.pass.cpp [hunk 1]
```diff
@@ -19,6 +19,11 @@
 
 #include "test_macros.h"
 
+#if defined(_MSC_VER)
+# pragma warning( disable: 4307 )
```
Comment (griwes) [inline .upstream-tests/test/std/utilities/time/time.cal/time.cal.day/time.cal.day.members/plus_minus_equal.pass.cpp:23]: Can you add comments to these, similarly how the 4244 disables have comments?

File: include/cuda/std/detail/libcxx/include/chrono [hunk 2]
```diff
@@ -2534,7 +2534,7 @@ year_month_day::__from_days(days __d) noexcept
     const unsigned __doy = __doe - (365 * __yoe + __yoe/4 - __yoe/100);              // [0, 365]
     const unsigned __mp = (5 * __doy + 2)/153;                                       // [0, 11]
     const unsigned __dy = __doy - (153 * __mp + 2)/5 + 1;                            // [1, 31]
-    const unsigned __mth = __mp + (__mp < 10 ? 3 : -9);                              // [1, 12]
+    const unsigned __mth = __mp + static_cast<unsigned>(__mp < 10 ? 3 : -9);       // [1, 12]
```
Comment (griwes) [inline include/cuda/std/detail/libcxx/include/chrono:2537]: Re-align the trailing comment with the 9nes in the previous lines.

File: include/cuda/std/detail/libcxx/include/chrono [hunk 3]
```diff
@@ -2553,9 +2553,9 @@ inline constexpr days year_month_day::__to_days() const noexcept
     const unsigned __dy  = static_cast<unsigned>(__d);
 
     const int      __era = (__yr >= 0 ? __yr : __yr - 399) / 400;
-    const unsigned __yoe = static_cast<unsigned>(__yr - __era * 400);                // [0, 399]
-    const unsigned __doy = (153 * (__mth + (__mth > 2 ? -3 : 9)) + 2) / 5 + __dy-1;  // [0, 365]
-    const unsigned __doe = __yoe * 365 + __yoe/4 - __yoe/100 + __doy;                // [0, 146096]
+    const unsigned __yoe = static_cast<unsigned>(__yr - __era * 400);                  // [0, 399]
+    const unsigned __doy = static_cast<unsigned>((153 * (__mth + static_cast<unsigned>(__mth > 2 ? -3 : 9)) + 2) / 5 + __dy-1);  // [0, 365]
```
Comment (griwes) [inline include/cuda/std/detail/libcxx/include/chrono:2557]: Why is the outer static_cast needed here?

**Review by griwes — CHANGES_REQUESTED** [review]

File: include/cuda/std/detail/libcxx/include/chrono (hunk 3 above)
Comment (wmaxey) [reply include/cuda/std/detail/libcxx/include/chrono:2557]: Two problems mainly. `__mth + negative signed integer` is a warning. The assignment ends up being a demotion as well after `xy / 5 + __dy-1`

File: include/cuda/std/detail/libcxx/include/chrono (hunk 2 above)
Comment (wmaxey) [reply include/cuda/std/detail/libcxx/include/chrono]: Realigned.

File: .upstream-tests/test/std/utilities/time/time.cal/time.cal.day/time.cal.day.members/plus_minus_equal.pass.cpp (hunk 1 above)
Comment (wmaxey) [reply .upstream-tests/test/std/utilities/time/time.cal/time.cal.day/time.cal.day.members/plus_minus_equal.pass.cpp:25]: Added comments.

File: include/cuda/std/detail/libcxx/include/chrono [hunk 4]
```diff
@@ -2554,7 +2554,8 @@ inline constexpr days year_month_day::__to_days() const noexcept
 
     const int      __era = (__yr >= 0 ? __yr : __yr - 399) / 400;
     const unsigned __yoe = static_cast<unsigned>(__yr - __era * 400);                  // [0, 399]
-    const unsigned __doy = static_cast<unsigned>((153 * (__mth + static_cast<unsigned>(__mth > 2 ? -3 : 9)) + 2) / 5 + __dy-1);  // [0, 365]
+    const unsigned __doy = static_cast<unsigned>(
+        (153 * (__mth + static_cast<unsigned>(__mth > 2 ? -3 : 9)) + 2) / 5 + __dy-1); // [0, 365]
```
Comment (wmaxey) [inline include/cuda/std/detail/libcxx/include/chrono:2558]: @griwes Thoughts on formatting this to align?

Comment (griwes) [reply include/cuda/std/detail/libcxx/include/chrono:2558]: That looks fine to me.

**Review by griwes — APPROVED** [review]

**Pull request merged**